jetty-ee9/jetty-ee9-websocket/jetty-ee9-websocket-jetty-client/pom.xml

```diff
@@ -16,6 +16,16 @@
 
   <build>
     <plugins>
+      <plugin>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <configuration>
+          <argLine>
+            @{argLine} ${jetty.surefire.argLine}
+            --add-modules org.eclipse.jetty.ee9.webapp
+            --add-reads org.eclipse.jetty.ee9.websocket.jetty.client=org.eclipse.jetty.ee9.webapp
```

Comment:
We don't declare an outright `requires` in the `module-info.java` we use `requires static` because webapp is optional for the client.

I think there would be no issue if we just had  `requires transitive org.eclipse.jetty.ee10.webapp;` in the module info, but we use `requires static transitive`.

Reply:
@lachlan-roberts you do not need and should not have the `reads` line in pom.xml. 

The `transitive` in `module-info` is not necessary either - you do not expose any methods in your module that expose any webapp classes. If in fact you delete `transitive`,  the distribution tests work fine.

The question is: do you need the `add-modules=org.eclipse.jetty.ee9.webapp` in your `pom.xml` when your `module-info` has declared it to be `requires static`. Normally, the `requires static` would mean that you don't need it to be present. _However_ because you have declared a `provides` of a class that depends upon classes in `org.eclipse.jetty.ee9.webapp` I believe that `jpms` insists that that module is always present. In other words, even if you are just deploying the client outside of jetty, with no webapps, you will always have to deploy the `org.eclipse.jetty.ee9.webapp` module.

I think the only way to solve that (ie to make it truly optional to provide `org.eclipse.jetty.ee9.webapp`) would be to make a separate `jetty-ee9-websocket-jetty-client-webapp` module that declares `requires` dependencies on both `org.eclipse.jetty.ee9.webapp` and `org.eclipse.jetty.ee9.websocket.client`  and `provides` the configuration class. That way, users who are deploying the client and don't care about webapps will deploy `org.eclipse.jetty.ee9.websocket.client`, but those that are deploying in jetty will deploy `org.eclipse.jetty.websocket.client.webapp`.